src/hipscat/pixel_tree/negative_tree.py

```diff
@@ -0,0 +1,22 @@
+from typing import List
+
+from hipscat.pixel_math import HealpixPixel
+from hipscat.pixel_tree import PixelAlignmentType, align_trees
+from hipscat.pixel_tree.pixel_tree import PixelTree
+
+
+def compute_negative_tree_pixels(tree: PixelTree) -> List[HealpixPixel]:
+    """Computes a 'negative pixel tree' consisting of the pixels needed to cover the full sky not in the tree
+
+    Args:
+        tree (PixelTree): the input tree to compute the negative of
+
+    Returns (List[HealpixPixel]):
+        A list of HEALPix pixels needed to cover the part of the sky not covered by the tree, using the least
+        number of pixels possible.
+    """
+    tree_pixels = set(tree.get_healpix_pixels())
+    full_tree = PixelTree.from_healpix([(0, i) for i in range(12)])
+    alignment = align_trees(full_tree, tree, alignment_type=PixelAlignmentType.OUTER)
+    aligned_tree = alignment.pixel_tree
+    return [p for p in aligned_tree.get_healpix_pixels() if p not in tree_pixels]
```

Comment:
That's a good question. I figured this was the easier way to get a list, and this way or a set difference are both O(n) operations, but set difference might have a more optimized implementation that doesn't involve python looping. 

I did a bit of testing, and seems like set difference doesn't do anything clever, so they're both about the same, with this way slightly quicker because it takes some time to construct the set then do the set difference. There might be some compiled set implementation out there like numpy does for arrays, but I don't think it's a bottleneck worth looking into a new dependency for. Here's my basic timing testing for 10,000 and 100,000 random ints:

10,000:
Set difference: 595 µs ± 6.39 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
List comprehension: 408 µs ± 757 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

100,000:
Set diff: 7.69 ms ± 85 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
List comp: 5.86 ms ± 32.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)